Source/Urho3D/IO/FileSystem.cpp

```diff
@@ -1067,4 +1067,34 @@ bool IsAbsolutePath(const String& pathName)
     return false;
 }
 
+String FileSystem::GetTemporaryDir() const

```

Comment:
Ok. Now I understood more what you wants for this new method. To me, this method implementation is too convoluted just to return a platform-specific temp dir. The fallback "./temp" directory name is of no use if it is not exists also. For all *NIX, the tmp dir should be:
- The environment variable TMPDIR, if it is defined.
- The value of the P_tmpdir macro, if it is defined. Note that importing the `stdio.h` still does not guarantee this.
- The last fallback is `/tmp`.

Reply:
> For all *NIX, the tmp dir should be

Are you sure that will also work on Android and iOS?

Reply:
I didn't say that. The embedded platform is very close to Linux but it is not POSIX compliance, so they probably have its own system call to get the temp dir, similar to how Windows platform has one.

Reply:
Mobile platforms are a reason why my implementation was as it was - it was unclear to me how exactly they behave regarding temp dir. My android has `TMPDIR` set and it points to `/data/local/tmp` so that will probably work. Not sure what happens on iOS. No device to test.